lib/screens/events/events.dart

```diff
@@ -28,29 +27,55 @@ class EventsScreen extends StatelessWidget {
             }
           },
           builder: (context, state) {
-            if (state is EventsInitial) {
-              return _buildInitial();
-            } else if (state is EventsSuccess) {
-              return _buildSuccess(context, state.json);
-            } else if (state is EventsLoading) {
-              return _buildLoading();
-            } else {
-              Log.e(tag: "EventsState", message: "State now is EventsError reached");
-              return Container(); // TODO the user should be shown the error on screen instead of a snackbar, and a retry button.
-            }
+            return Scaffold(
+              resizeToAvoidBottomInset: false,
+              extendBodyBehindAppBar: true,
+              appBar: AppBar(
+                elevation: 0,
+                backgroundColor: Colors.transparent,
+                leading: Container(
+                  padding: EdgeInsets.only(left: D.horizontalPadding - 10, top: 10),
+                  child: IconButton(
+                    icon: Icon(Icons.arrow_back_ios, color: Colors.white, size: 30),
+                    onPressed: () => Navigator.of(context).pop(),
+                  ),
+                ),
+              ),
+              body: Stack(
+                children: [
+                  ScreenBackground(elementId: 0),
+                  if (state is EventsInitial)
+                    _buildLoading(context)
+                  else if (state is EventsSuccess)
+                    _buildSuccess(context, state.json)
+                  else if (state is EventsLoading)
+                    _buildLoading(context)
+                  else
+                    _buildAskReload()
+                ],
+              ),
+            );
           },
         ),
       ),
     );
   }
 
-  Widget _buildLoading() {
-    return LoadingScreen();
+  Widget _buildAskReload() {
+    //TODO: Implement a Screen to reload
+    return Container();
   }
 
-  Widget _buildSuccess(BuildContext context, List<Event> eventsList) {
-    //TODO On success UI
+  Widget _buildLoading(BuildContext context) {
+    double width = MediaQuery.of(context).size.width;
+    return Stack(
+      children: [
+        Center(child: ECellLogoAnimation(size: width / 2)),
+      ],
+    );
```

Comment:
Single widget in stack can be replaced with just the element widget. Do the changes in other screens too.
```suggestion
    return Center(child: ECellLogoAnimation(size: width / 2));
```